From c73175f52bd156e6ce15581a55eb8a1ebfec5900 Mon Sep 17 00:00:00 2001
From: "Yann E. MORIN" <yann.morin.1998@free.fr>
Date: Tue, 25 Dec 2018 10:36:32 +0100
Subject: [PATCH] buildsys: relax dependencies to be able to just build
 systemd-boot

Selectively remove the checks one by one until we are able to build
systemd-boot alone.

The add_languages() trick was provided by James:
    https://github.com/systemd/systemd/pull/11279

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>
---
 meson.build          | 9 ++++-----
 src/test/meson.build | 2 +-
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/meson.build b/meson.build
index b338886c3..337e0145e 100644
--- a/meson.build
+++ b/meson.build
@@ -287,8 +287,7 @@ want_tests = get_option('tests')
 slow_tests = want_tests != 'false' and get_option('slow-tests')
 install_tests = get_option('install-tests')
 
-cxx = find_program('c++', required : fuzzer_build)
-if cxx.found()
+if add_languages('cpp', required : fuzzer_build)
         #  Used only for tests
         add_languages('cpp')
         cxx_cmd = ' '.join(meson.get_compiler('cpp').cmd_array())
@@ -660,7 +659,7 @@ conf.set('GPERF_LEN_TYPE', gperf_len_type,
 ############################################################
 
 if not cc.has_header('sys/capability.h')
-        error('POSIX caps headers not found')
+        warning('POSIX caps headers not found')
 endif
 foreach header : ['crypt.h',
                   'linux/btrfs_tree.h',
@@ -871,10 +870,10 @@ libcrypt = cc.find_library('crypt')
 libcap = dependency('libcap', required : false)
 if not libcap.found()
         # Compat with Ubuntu 14.04 which ships libcap w/o .pc file
-        libcap = cc.find_library('cap')
+        libcap = cc.find_library('cap', required : false)
 endif
 
-libmount = dependency('mount',
+libmount = dependency('mount', required : false,
                       version : fuzzer_build ? '>= 0' : '>= 2.30')
 
 want_seccomp = get_option('seccomp')
diff --git a/src/test/meson.build b/src/test/meson.build
index ea049a6fb..d9d87e02c 100644
--- a/src/test/meson.build
+++ b/src/test/meson.build
@@ -957,7 +957,7 @@ tests += [
 
 ]
 
-if cxx.found()
+if cxx_cmd != ''
         tests += [
                 [['src/libsystemd/sd-bus/test-bus-vtable-cc.cc'],
                  [],
-- 
2.14.1

